src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1369,9 +1369,18 @@ private static bool IsCompilerAnalyzer(DiagnosticAnalyzer analyzer)
 
         public void Dispose()
         {
-            this.CompilationEventQueue?.TryComplete();
-            this.DiagnosticQueue?.TryComplete();
-            _queueRegistration.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
>protected virtual void Dispose(bool disposing) [](start = 8, length = 46)

I am not sure why this method is added. It looks like no one overrides it, there is only one call site and we always pass true.